scripts/ci/errno.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2021 Nordic Semiconductor NA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Check minimal libc error numbers against newlib.
+
+This script loads the errno.h included in Zephyr's minimal libc and checks its
+contents against the SDK's newlib errno.h. This is done to ensure that both C
+libraries are aligned at all times.
+"""
+
+
+import os
+from pathlib import Path
+import re
+import sys
+
+def parse_errno(path):
+    with open(path, 'r') as f:
+        r = re.compile(r'^\s*#define\s+([A-Z]+)\s+([0-9]+)')
+        errnos = []
+        for line in f:
+            m = r.match(line)
+            if m:
+                errnos.append((m.group(1), m.group(2)))
```

Comment:
nit: `errnos.append(m.groups())`

Reply:
Not sure if there are more groups defined by a regex match, so I'd rather be explicit here.

Reply:
only 2 are defined in the match expression, so we can't have more

Reply:
Actually, this doesn't work because the first element in `m.group()` is [the entire match](https://docs.python.org/3/library/re.html#match-objects)

Reply:
Hmm... I get this:
```
In [7]: import re

In [8]: m = re.match(r'^\s*#define\s+([A-Z]+)\s+([0-9]+)', '#define X 0')

In [9]: m.groups()
Out[9]: ('X', '0')
```

Reply:
Doh, I was using `m.group()` and not `m.groups()`. Fixed now @gmarull